corehq/ex-submodules/casexml/apps/phone/cleanliness.py

```diff
@@ -0,0 +1,102 @@
+from collections import namedtuple
+from datetime import datetime
+from casexml.apps.case.models import CommCareCase
+from casexml.apps.case.util import get_indexed_case_ids, get_reverse_indexed_case_ids, get_open_case_ids, \
+    get_closed_case_ids, get_indexed_cases
+from casexml.apps.phone.models import OwnershipCleanlinessFlag
+
+
+FootprintInfo = namedtuple('FootprintInfo', ['base_ids', 'all_ids'])
+CleanlinessFlag = namedtuple('CleanlinessFlag', ['is_clean', 'hint'])
+
+
+def set_cleanliness_flags_for_domain(domain):
+    """
+    Sets all cleanliness flags for an entire domain.
+    """
+    for owner_id in get_all_case_owner_ids(domain):
+        set_cleanliness_flags(domain, owner_id)
+
+
+def get_all_case_owner_ids(domain):
+    """
+    Get all owner ids that are assigned to cases in a domain.
+    """
+    key = ["all owner", domain]
+    submitted = CommCareCase.get_db().view('case/all_cases',
+        group_level=3,
+        startkey=key,
+        endkey=key + [{}],
+    ).all()
+    return set([row['key'][2] for row in submitted])
```

Comment:
How would you feel about instituting a new convention that all (new) calls to couchdb views need to go into a file called `dbaccessors.py` in the app corresponding to the design doc it's referencing (so in this case `casexml.apps.case.dbaccessors`)? The benefits of this are:

- the app itself more or less tightly controls the interface through which callers access the database
- all of these functions are in one place, making it easy to see if they can be combined or replaced, or if there are duplicates
- it's a first step towards being able to rewrite how we interface with our data—whether that's consolidating and removing couch views or changing the storage and querying backend altogether
- it checks out against with my gut as being a "nice" organizational pattern anyway